server/journeys/referral/details/schemas.ts

```diff
@@ -33,15 +37,28 @@ export const schemaFactory =
       .object({
         incidentDate: z
           .string({ message: INCIDENT_DATE_MSG })
-          .superRefine((val, ctx) => {
-            const result = parseDate(val)
-            if (!result.success) {
-              ctx.addIssue({ code: z.ZodIssueCode.custom, message: INCIDENT_DATE_MSG })
-            } else if (result.data! > todayString()) {
-              ctx.addIssue({ code: z.ZodIssueCode.custom, message: INCIDENT_DATE_FUTURE_MSG })
-            }
+          .min(8, { message: INCIDENT_DATE_MSG })
+          .max(10, { message: INCIDENT_DATE_MSG })
+          .transform(value => value.split(/[-/]/).reverse())
+          .transform(value => {
+            // Prefix month and date with a 0 if needed
+            const month = value[1]?.length === 2 ? value[1] : `0${value[1]}`
+            const date = value[2]?.length === 2 ? value[2] : `0${value[2]}`
+            return `${value[0]}-${month}-${date}T00:00:00Z` // We put a full timestamp on it so it gets parsed as UTC time and the date doesn't get changed due to locale
```

Comment:
Are there any existing tests for the date parsing?

Reply:
For the different edge cases? There are a couple in the tests.test.ts file, and I added more in the e2e PR I've done.